### PR TITLE
Independent Set Generator

### LINKED ISSUE #27: Instance generator for Indepentend Set
Input is a nx.Graph. For MIP formulation see e.g. [here](http://www.princeton.edu/~aaa/Public/Teaching/ORF523/S16/ORF523_S16_Lec11_gh.pdf).

### LINKED ISSUE #28: Independent Set Param Generator Erdös-Renyi
Create input for #27 as in [this](https://arxiv.org/pdf/1906.01629.pdf) paper.

### LINKED ISSUE #29: Independent Set Param Generator Barabasi
Create input for #27 following VII.A of [this](https://journals.aps.org/rmp/pdf/10.1103/RevModPhys.74.47) paper. Check if networkx supports those graphs already.

### PR DESCRIPTION
**Pull request by mmghannam**
Closes #27, closes #28, closes #29

### CODE REVIEW

File: geco/mips/independent_set.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        pyscipopt model of the generated instance
+
+    References
+    ----------
+    .. [1] https://www.princeton.edu/~aaa/Public/Teaching/ORF523/S16/ORF523_S16_Lec11_gh.pdf
+    """
+    model = scip.Model(name)
+
+    vars = {
+        str(node): model.addVar(lb=0, ub=1, obj=1, name=str(node), vtype="B")
```
Comment (CharJon) [inline geco/mips/independent_set.py]: I think I remember limitations to names for variables in ".lp" files. So just calling str() might be dangerous.
What about calling `networkx.relabel.convert_node_labels_to_integers` first.

File: geco/mips/independent_set.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        for node in graph.nodes
+    }
+
+    for u, v in graph.edges:
+        model.addCons(vars[str(u)] + vars[str(v)] <= 1)
```
Comment (CharJon) [inline geco/mips/independent_set.py]: See above.

File: geco/mips/independent_set.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    model.setMaximize()
+
+    return model
+
+
+def _get_cliques(graph: nx.Graph) -> list:
+    """
+    Partition the graph into cliques using a greedy algorithm, this code is
```
Comment (CharJon) [inline geco/mips/independent_set.py]: Does networkx offer something similar?

**Review by CharJon — CHANGES_REQUESTED** [review]

File: geco/mips/independent_set.py (hunk 3 above)
Comment (mmghannam) [reply geco/mips/independent_set.py]: [Yes](https://networkx.org/documentation/stable/reference/algorithms/clique.html), but I wanted to be as close as possible to the original implementation.

**Pull request merged**